legion/engine/audio/systems/audiosystem.inl

```diff
@@ -276,8 +277,8 @@ namespace legion::audio
     {
         log::debug("Destroying Audio Listener...");
 
-        listenerCount = math::max((int)(listenerCount - 1), 0);
-        if (listenerCount == 0)
+        listenerCount = math::max(static_cast<int>(listenerCount - 1), 0);
+        if (listenerCount == 0 && scenemanagement::SceneManager::doNotCreateEntities == false)
         {
             log::debug("No Listeners left, resetting listener");
             m_listenerEnt = ecs::entity_handle();
```

Comment:
honestly, audio system should just never create entities, but instead, log a warning that there are no listeners.

Reply:
this is more of a problem to fix for jelle than for me. I dont know much about audiosystem. But if we remove this entity destruction breaks completely

Reply:
jelle and I fixed it